R/print.hdi.R

```diff
@@ -0,0 +1,104 @@
+#' @export
+print.hdi <- function(x, digits = 2, ...) {
+  .print_hdi(x, digits, title = "Highest Density Interval", ci_string = "HDI", ...)
+}
+
+
+.print_hdi <- function(x, digits, title, ci_string, ...) {
+  insight::print_color("blue", sprintf(
+    "# %s%s\n\n",
+    title,
+    ifelse(all(x$CI[1] == x$CI), "", "s")
+  ))
+
+  ci <- unique(x$CI)
+
+  # find the longest HDI-value, so we can align the brackets in the ouput
+  x$CI_low <- sprintf("%.*f", digits, x$CI_low)
+  x$CI_high <- sprintf("%.*f", digits, x$CI_high)
+
+  maxlen_low <- max(nchar(x$CI_low))
+  maxlen_high <- max(nchar(x$CI_high))
+
+  x$HDI <- sprintf("[%*s %*s]", maxlen_low, x$CI_low, maxlen_high, x$CI_high)
+
+  if (length(ci) == 1) {
+    xsub <- .remove_column(x, c("CI", "CI_low", "CI_high"))
+    colnames(xsub)[ncol(xsub)] <- sprintf("%i%% %s", ci, ci_string)
+    print_data_frame(xsub, digits = digits)
+  } else {
+    for (i in ci) {
+      xsub <- x[x$CI == i, -which(colnames(x) == "CI"), drop = FALSE]
+      xsub <- .remove_column(xsub, c("CI", "CI_low", "CI_high"))
+      colnames(xsub)[ncol(xsub)] <- sprintf("%i%% %s", i, ci_string)
+      print_data_frame(xsub, digits = digits)
+      cat("\n")
+    }
+  }
+}
+
+
+
+
+
+
```

Comment:
I see you like empty lines... ;-)

Reply:
haha... I guess so, it helps me see the different functions (especially when navigating through internals). An alternative could be adding  `#-----------------` between functions?